src/AspNet.Security.OAuth.BattleNet/BattleNetAuthenticationOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http;
+using System;
+
+namespace AspNet.Security.OAuth.BattleNet {
+    /// <summary>
+    /// Defines a set of options used by <see cref="BattleNetAuthenticationHandler"/>.
+    /// </summary>
+    public class BattleNetAuthenticationOptions : OAuthAuthenticationOptions {
+        public BattleNetAuthenticationOptions() {
+            AuthenticationScheme = BattleNetAuthenticationDefaults.AuthenticationScheme;
+            Caption = BattleNetAuthenticationDefaults.Caption;
+            ClaimsIssuer = BattleNetAuthenticationDefaults.Issuer;
+
+            CallbackPath = new PathString(BattleNetAuthenticationDefaults.CallbackPath);
+
+            this.SetServerRegion(ServerRegion.US); //Default to US
+
+            UserInformationEndpoint = BattleNetAuthenticationDefaults.UserInformationEndpoint;
+
+            SaveTokensAsClaims = false;
+        }
+
+        public void SetServerRegion(ServerRegion serverRegion) {
```

Comment:
I would argue that it would be confusing to have a property change other properties. There is no good way for people to easily see that they would have to either set one property (ServerRegion) or set two properties (the endpoints) and not all 3. We wouldn't want to just leave the one property (ServerRegion) because that wouldnt be consistent with the other ones and takes away some customization.  Also not having a helper method there might be ok so infact to be less confusing and more consistant I can just remove it all together.  I just want a confusing senario where a user is setting 3 properties for no purpose.